api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java

```diff
@@ -297,6 +444,24 @@ default Metadata getMetadata() {
         }
     }
 
+    /**
+     * Acknowledge negatively this message.
+     * <code>nack</code> is used to indicate that the processing of a message failed. The reason is passed as parameter.
+     *
+     * @param reason the reason of the nack, must not be {@code null}
+     * @return a completion stage completed when the message is negative-acknowledgement has completed. If the
+     *         negative acknowledgement fails, the completion stage propagates the failure.
+     */
+    @Experimental("nack support is a SmallRye-only feature")
+    default CompletionStage<Void> nack(Throwable reason) {
+        Function<Throwable, CompletionStage<Void>> nack = getNack();
+        if (nack == null) {
+            return CompletableFuture.completedFuture(null);
```

Comment:
Has the throwable been logged anywhere prior to this point?

Would it be beneficial to log out the throwable as we're swallowing it? How would a developer know something went wrong without a configured `nack`?

Reply:
That's a good point. When you create an instance of Message, you should pass the `nack` and `ack` function, but I agree. If by any chance you use this default implementation, we should log the reason (if there is no nack function)

Reply:
Fixed.